app/controllers/application_controller/automate.rb

```diff
@@ -115,6 +108,24 @@ def resolve_button_reset_or_none
   end
   private :resolve_button_reset_or_none
 
+  def resolve_react
```

Comment:
I think its best to change the name `resolve_react` to a different name which aligns with our feature.